File: Doc/examples/python/send_logs.py
#!/usr/bin/env python3
import json
import os
import sys
import urllib.error
import urllib.request


def send_log(base_url, payload):
    request = urllib.request.Request(
        f"{base_url}/logs",
        data=json.dumps(payload).encode("utf-8"),
        headers={"Content-Type": "application/json"},
        method="POST",
    )

    with urllib.request.urlopen(request, timeout=5) as response:
        body = response.read().decode("utf-8")
        print(f"POST /logs -> {response.status} {body}")


def fetch_incidents(base_url):
    request = urllib.request.Request(f"{base_url}/incidents")
    with urllib.request.urlopen(request, timeout=5) as response:
        body = response.read().decode("utf-8")
        print(f"GET /incidents -> {response.status} {body}")


def main():
    base_url = os.environ.get("BASE_URL", "http://localhost:8080")
    logs = [
        {"service": "payment-api", "level": "error", "message": "Database connection failed"},
        {"service": "payment-api", "level": "error", "message": "Timeout while querying users"},
        {"service": "payment-api", "level": "error", "message": "Retry budget exhausted"},
    ]

    try:
        for payload in logs:
            send_log(base_url, payload)
        fetch_incidents(base_url)
    except urllib.error.URLError as exc:
        print(f"request failed: {exc}", file=sys.stderr)
        raise SystemExit(1)


if __name__ == "__main__":
    main()

File: Doc/examples/node/send_logs.mjs
#!/usr/bin/env node

const baseUrl = process.env.BASE_URL || "http://localhost:8080";

const logs = [
  { service: "payment-api", level: "error", message: "Database connection failed" },
  { service: "payment-api", level: "error", message: "Timeout while querying users" },
  { service: "payment-api", level: "error", message: "Retry budget exhausted" },
];

async function sendLog(payload) {
  const response = await fetch(`${baseUrl}/logs`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  const body = await response.text();
  console.log(`POST /logs -> ${response.status} ${body}`);
}

async function fetchIncidents() {
  const response = await fetch(`${baseUrl}/incidents`);
  const body = await response.text();
  console.log(`GET /incidents -> ${response.status} ${body}`);
}

for (const payload of logs) {
  await sendLog(payload);
}

await fetchIncidents();
